### PR TITLE
increase resiliency of new overlay feature

### PR DESCRIPTION
**Pull request by ozdanborne**
## Description

The new overlay feature was implemented such that other controllers pull the computed installation spec from the status field. However, when run on systems which do not have the updated CRD definition, this was always nil.

This PR addresses this problem by refactoring `GetInstallation` (the primary function used by other controllers to get current installation state) to compute the final installation spec instead of pulling it from Status.

## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

File: pkg/controller/logstorage/logstorage_controller.go [hunk 1]
```diff
@@ -318,7 +318,7 @@ func (r *ReconcileLogStorage) Reconcile(request reconcile.Request) (reconcile.Re
 		r.status.OnCRFound()
 	}
 
-	installationCR, err := installation.GetInstallation(context.Background(), r.client)
+	variant, installationCR, err := installation.GetInstallation(context.Background(), r.client)
```
Comment (tmjd) [inline pkg/controller/logstorage/logstorage_controller.go:321]: Same here, `installationCR` isn't a correct name anymore.

File: pkg/controller/logstorage/logstorage_controller.go [hunk 2]
```diff
@@ -466,7 +466,7 @@ func (r *ReconcileLogStorage) Reconcile(request reconcile.Request) (reconcile.Re
 	if ls != nil {
 		hdler = utils.NewComponentHandler(log, r.client, r.scheme, ls)
 	} else {
-		hdler = utils.NewComponentHandler(log, r.client, r.scheme, installationCR)
+		hdler = utils.NewComponentHandler(log, r.client, r.scheme, managementClusterConnection)
```
Comment (tmjd) [inline pkg/controller/logstorage/logstorage_controller.go:469]: Are you getting input from Brian or Rene about using this for the owner ref?

File: pkg/controller/logstorage/logstorage_controller.go [hunk 3]
```diff
@@ -501,7 +501,7 @@ func (r *ReconcileLogStorage) Reconcile(request reconcile.Request) (reconcile.Re
 
 	component := render.LogStorage(
 		ls,
-		installationCR.Status.Computed,
+		installationCR,
```
Comment (tmjd) [inline pkg/controller/logstorage/logstorage_controller.go:504]: Name again

**Review by tmjd — COMMENTED** [review]
A few nits on some names.
I wouldn't be upset if you normalized all the names we use for the installation/network/installationCR variable name we use in the different controllers. But my request is to at least update the ones called `installationCR`.

File: pkg/controller/logstorage/logstorage_controller.go (hunk 2 above)
Comment (Brian-McM) [reply pkg/controller/logstorage/logstorage_controller.go:469]: This seems like the right thing to do.

**Review by tmjd — APPROVED** [review]
LGTM, please squash before merging.

**Pull request merged**